### ISSUE TITLE
CMake build

### ISSUE THREAD

**Issue by KOLANICH** [google/lyra#50]
It would be nice if all the components of the software that is not already present in the distro on user's mavhine be buildable with CMake, and ones that are presente just linked.

**Comment by aluebs** [comment]
Thank you for the request. Do you mean to use cmake instead of bazel?

**Comment by KOLANICH** [comment]
I mean exactly this. I am not a big fan of bazel (after my poor experience with it a few years ago when I tried to build TensorFlow with it). I cannot exclude that either I don't understand bazel enough, or that it may be misconfigured, but the 4 things that I currently perceive in bazel as the worst are

* it unneedingly builds dependencies that are already prebuilt by third parties and installed in the system
* it doesn't have CPack-like module to generate packages I can install into the system
* it is large itself, because its dependencies are embedded into it, including Java machine. It is not the way software is meant to be distributed for installation into distros. For installation into distros software should rely on the deps provided by distro and on distro package manager to install them.
* It depends on JVM which is a large dependency, which is not very friendly for CI to install it every time.

In other words, Bazel was created the way assumming that its users is a big rich enterprise that can afford itself to waste resources, like Google is, and ignoring needs of everyone else, especially the needs of just users of distros who just want to build a bleeding edge version of the software from source as faster and as cheaper as possible, generate the package that reuses already installed deps as mjch as possible and install it into the system and publish it to allow other people to install it.

**Comment by aluebs** [comment]
Thank you for the detailed explanation. Those are all great points. Unfortunately I don't see us moving away from bazel any time soon, since it simplifies quite a bit our workflows. And maintaining a second build system independently seems like a lot of work and bug-prone. Do you know if there is a good way to translate bazel files into CMake?

**Comment by KOLANICH** [comment]
I have just  googled a little and found these 2 projects:

* https://github.com/google/bazel-to-cmake
* https://github.com/lizan/bazel-cmakelists

I have used neither of them, but feel like I may need to give them a try somewhen for PlaidML   - a machine learning library using OpenCL, so suitable for AMDs. 

I can say some thing though about both of them. None of them has good enough source code, both look like they were made in haste with a lot of dirty shortcuts to spare the devs from writing a few lines of code, and both of them look a bit abandoned.

**Comment by aluebs** [comment]
If you end up trying any of them on Lyra, I'd love to hear about your experience.

**Comment by xnorpx** [comment]
dependencies is probably gonna be the painful part.

**Comment by CODER591** [comment]
Are there any of CMake files? I am not able to find anything related. The issue seems to be pretty old.
